clients/src/test/java/org/apache/kafka/common/record/ByteBufferLogInputStreamTest.java

```diff
@@ -120,4 +121,26 @@ public void iteratorRaisesOnTooLargeRecords() {
         logInputStream.nextBatch();
     }
 
+    @Test
+    public void testReadUnsignedIntFromInputStream() {
```

Comment:
Could you move the tests to ```ByteBufferInputStreamTest```? If there is no such test class, feel free to create new one.